pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -1185,6 +1185,7 @@ func initPublications(zedagentCtx *zedagentContext) {
 		log.Fatal(err)
 	}
 	getconfigCtx.pubCipherContext.ClearRestarted()
+	populateCipherContexts(getconfigCtx)
```

Comment:
Am I correct that here we expect pubsub status file to be persistent on disk and once we create a new publication, this file will be read and the `pubCipherContext` will be populated from it?

Reply:
Yep, the pub-sub state is persistent (stored on disk and survives system restart) when the Publication is created with the `Persistent: true` flag. And that's the case here. Nevertheless, I'm not sure the context is restored in a synchronous way, and we don't need to wait for any notifications...

Reply:
@OhmSpectator when we call NewPublication (or NewSubscription for that matter) for a persistent publication (or subscription), the pubsub package fills in the items from the persistent storage. Thus after those one can do a GetAll() etc.

We have the same pattern in nim for the DevicePortConfigList.